src/Components/Web.JS/src/Services/NavigationUtils.ts

```diff
@@ -47,6 +47,32 @@ export function isWithinBaseUriSpace(href: string) {
   && (nextChar === '' || nextChar === '/' || nextChar === '?' || nextChar === '#');
 }
 
+export function isSamePageWithHash(absoluteHref: string): boolean {
+  const hashIndex = absoluteHref.indexOf('#');
+  return hashIndex > -1 && location.href.replace(location.hash, '') === absoluteHref.substring(0, hashIndex);
+}
+
+export function performScrollToElementOnTheSamePage(absoluteHref : string): void {
+  const hashIndex = absoluteHref.indexOf('#');
+  if (hashIndex === absoluteHref.length - 1) {
+    return;
+  }
+
+  const identifier = absoluteHref.substring(hashIndex + 1);
+  scrollToElement(identifier);
+}
+
+export function scrollToElement(identifier: string): boolean {
+  const element = document.getElementById(identifier);
+
+  if (element) {
+    element.scrollIntoView();
+    return true;
+  }
+
+  return false;
```

Comment:
Is the return value used somewhere? Just reading this PR it doesn't appear to be.

I know you're inheriting existing code here, but while it's being looked on it looks like it could be a lot more streamlined, e.g.:

```ts
export function scrollToElement(identifier: string) {
    document.getElementById(identifier)?.scrollIntoView();
}
```